packages/pymedphys_dicom/src/pymedphys_dicom/shim.py

```diff
@@ -0,0 +1,14 @@
+class Shim:
+    def __init__(self, import_error):
+        self.import_error = import_error
+
+    def __getattribute__(self, name):
+        raise self.import_error
+
+
+try:
+    import pydicom
+
+    pydicom.config.enforce_valid_values = True
+except ImportError as e:
+    pydicom = Shim(e)
```

Comment:
Once we remove all of the sub packages I thought the above method would be a good way to make packages optional.

Reply:
Yep, lovely :)

I'm not sure I understand the `Shim` class? Why not:

```python
try:
    import pydicom

    pydicom.config.enforce_valid_values = True
except ImportError:
    raise
```

Reply:
Essentially it makes it so the import error is raised when pydicom is actually attempted to be used instead of at the initial import time. It will need a bit of playing with to work out how best to implement it. But I imagined having a shim package which also has numpy and all other external packages. Then we just always import external packages via shim everywhere...

I think it'll be one of those things that will improve as we implement it I believe...